node/cn/peer_set_test.go

```diff
@@ -0,0 +1,654 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package cn
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/klaytn/klaytn/node"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+func TestPeerSet_Register(t *testing.T) {
+	peerSet := newPeerSet()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	cnPeer := NewMockPeer(mockCtrl)
+	cnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.CONSENSUSNODE)).Times(1)
+	cnPeer.EXPECT().GetAddr().Return(addr1).Times(3)
+	cnPeer.EXPECT().GetID().Return(nodeID1.String()).Times(3)
+	cnPeer.EXPECT().Broadcast().AnyTimes()
+
+	pnPeer := NewMockPeer(mockCtrl)
+	pnPeer.EXPECT().ConnType().Return(p2p.ConnType(node.PROXYNODE)).Times(1)
+	pnPeer.EXPECT().GetAddr().Return(addr2).Times(3)
+	pnPeer.EXPECT().GetID().Return(nodeID2.String()).Times(3)
+	pnPeer.EXPECT().Broadcast().AnyTimes()
+
+	enPeer := NewMockPeer(mockCtrl)
+	enPeer.EXPECT().ConnType().Return(p2p.ConnType(node.ENDPOINTNODE)).Times(1)
+	enPeer.EXPECT().GetAddr().Return(addr3).Times(3)
+	enPeer.EXPECT().GetID().Return(nodeID3.String()).Times(3)
+	enPeer.EXPECT().Broadcast().AnyTimes()
+
+	assert.NoError(t, peerSet.Register(cnPeer))
+	assert.NoError(t, peerSet.Register(pnPeer))
+	assert.NoError(t, peerSet.Register(enPeer))
+
```

Comment:
How about ensuring peer registeration by checking the increment of `len(ps.peers)`?

Reply:
I saw that it is tested on `TestPeerSet_Peer_And_Len`. 
Now, I understand that this test is for registration failure cases.